components/brave_wallet/common/hex_utils.cc

```diff
@@ -148,18 +148,32 @@ bool HexValueToInt256(std::string_view hex_input, int256_t* out) {
 
 std::string Uint256ValueToHex(uint256_t input) {
   std::string result;
-  result.reserve(32);
-
-  static constexpr char kHexChars[] = "0123456789abcdef";
-  while (input) {
-    uint8_t i = static_cast<uint8_t>(input & static_cast<uint256_t>(0x0F));
-    UNSAFE_TODO(result.insert(result.begin(), kHexChars[i]));
-    input >>= 4;
+  result.reserve(34);
+  result.append("0x");
+
+  auto input_span = base::byte_span_from_ref(input);
+  bool skipping_zeros = true;
+  for (auto& byte : base::Reversed(input_span)) {
+    if (skipping_zeros && !byte) {
+      continue;
+    }
+    if (skipping_zeros && byte) {
+      skipping_zeros = false;
+      if (byte <= 0xf) {
+        std::string one_char_byte;
+        base::AppendHexEncodedByte(byte, one_char_byte, false);
+        result += one_char_byte[1];
+        continue;
+      }
+    }
+    base::AppendHexEncodedByte(byte, result, false);
```

Comment:
maybe we can simplify this a bit to:

```cxx
auto reversed_input = base::byte_span_from_ref(input);
auto last_zero = base::range::find_if(reversed_input, [](uint8_t byte){ return byte != 0; });
if (last_zero == reversed_input.end()) {
  return "0x0";
}

result = base::HexEncode(input.first(input.size() - std::distance(reversed_input.begin(), last_zero) -1);
```

What do you think?

Reply:
I'm actually having second thoughts about fiddling with the memory representation of this number, and assuming little endianness. I think the previous function was better because it nibbles the number into a byte, without assuming anything about the bit layout. 

I think we can keep it as it was and just change it to something like:
```
  static constexpr char kHexChars[] = "0123456789abcdef";
  while (input) {
    uint8_t i = static_cast<uint8_t>(input & static_cast<uint256_t>(0x0F));
    result.append(base::span(kHexChars)[i]);
    input >>= 4;
  }
```

Reply:
I was testing another way to do this and this also seems to work https://godbolt.org/z/z7jncxnTT

Reply:
This version nibbles the leading zeroes using a clang intrinsic https://godbolt.org/z/1hc65zrdo

Reply:
Thanks! The last one actually fails on "0x0001BEEF..." value.
I've changed implementation to use this intrinsic but still without indexes. Looks a bit more clearer.

Reply:
I have thought about this a bit further and I think we should use the bit shift, as I don't think it will be heavier than looping through the bytes. The compiler is smart enough to optimise it, and avoid the whole bit shift in the first place, as can be seen in the example below of the codegen produced.

https://godbolt.org/z/43s1WWvsn

Reply:
Thanks.